portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -63,53 +67,61 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
+      } catch (NumberFormatException e) {
+        System.err.println("Could not convert to int: " + user_num);
+        num_comments = -1;
+      }
 
-    try {
-      num_comments = Integer.parseInt(user_num);
-    } catch (NumberFormatException e) {
-      System.err.println("Could not convert to int: " + user_num);
-      num_comments = -1;
-    }
+      if (num_comments < 0 || num_comments > 15) {
+        response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        response.setContentType("application/json");
+        errorMap.put("type", "VALIDATION");
+        errorMap.put("message", "Please enter an integer between 0 and 15.");
+        response.getWriter().println(gson.toJson(errorMap));
+        return;
+      }
 
-    if (num_comments < 0 || num_comments > 15) {
-      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
-      response.setContentType("application/json");
-      response.getWriter().println(gson.toJson("Please enter an integer between 0 and 15."));
-      return;
-    }
+      Query commentQuery;
+      if (request.getParameter("order").equals("asc")) {
+        commentQuery = new Query("Comment").addSort(ENTITY_TIME_PARAM, SortDirection.ASCENDING);
+      } else {
+        commentQuery = new Query("Comment").addSort(ENTITY_TIME_PARAM, SortDirection.DESCENDING);
+      }
 
-    Query commentQuery;
-    if (request.getParameter("order").equals("asc")) {
-      commentQuery = new Query("Comment").addSort(ENTITY_TIME_PARAM, SortDirection.ASCENDING);
-    } else {
-      commentQuery = new Query("Comment").addSort(ENTITY_TIME_PARAM, SortDirection.DESCENDING);
-    }
+      PreparedQuery results = datastore.prepare(commentQuery);
 
-    PreparedQuery results = datastore.prepare(commentQuery);
+      List<Comment> comments = new ArrayList<>();
 
-    List<Comment> comments = new ArrayList<>();
+      int counter = 0;
+      for (Entity entity : results.asIterable()) {
+        if (counter >= num_comments) {
+          break;
+        }
+        String text = (String) entity.getProperty(ENTITY_TEXT_PARAM);
+        String name = (String) entity.getProperty(ENTITY_NAME_PARAM);
 
-    int counter = 0;
-    for (Entity entity : results.asIterable()) {
-      if (counter >= num_comments) {
-        break;
+        Comment comment = new Comment(text, name);
+        comments.add(comment);
+        counter++;
       }
-      String text = (String) entity.getProperty(ENTITY_TEXT_PARAM);
-      String name = (String) entity.getProperty(ENTITY_NAME_PARAM);
 
-      Comment comment = new Comment(text, name);
-      comments.add(comment);
-      counter++;
-    }
+      String json = gson.toJson(comments);
+      response.setContentType("application/json");
+      response.getWriter().println(json);
 
-    String json = gson.toJson(comments);
-    response.setContentType("appplication/json;");
-    response.getWriter().println(json);
+    } else {
+      response.setContentType("text/html");
```

Comment:
Please set an appropriate error code. `SC_UNAUTHORIZED` looks like the best fit.